src/GraphQL.Tests/Bugs/BubbleUpTheNullToNextNullable.cs

```diff
@@ -111,9 +111,15 @@ public void ListOfNonNull_containing_null_should_bubble_up_the_null()
             var errors = new[]
             {
                 new ExecutionError("Error trying to resolve field 'listOfNonNullable'.", new InvalidOperationException(
-                    "Cannot return a null member within a non-null list for list index 1."))
+                    "Cannot return null for non-null type. Field: listOfNonNullable, Type: String!."))
                 {
-                    Path = new object[] {"nonNullableDataGraph", "listOfNonNullable"}
+                    Path = new object[] {"nonNullableDataGraph", "listOfNonNullable", 1}
+                },
+
+                new ExecutionError("Error trying to resolve field 'listOfNonNullable'.", new InvalidOperationException(
```

Comment:
I see that parent `nonNullableDataGraph` is non-null too. The error occurred on the **element** of the array but error message tals about array field itself (OK, `Path` is a hint for us that in fact the error occurred on the element). My question is why we do not see the same error message for the parent ``nonNullableDataGraph` field (and even maybe parent of parent and so on)? And the second question - do we need to add an error related to the array **element** (not array itself)? I get the impression that now there is no clear zone of responsibility about which node should add an error and what error.

Reply:
The spec prevents us from adding errors to parents due to nulls bubbling up:

> If the field returns null because of an error which has already been added to the "errors" list in the response, the "errors" list must not be further affected. That is, only one error should be added to the errors list per field.

Reply:
At least, I think that's what it says.  There's a bunch of paragraphs on it, and it seems to me that we should be listing 1+ errors, then bubbling up the null as far as necessary.  Which is what currently happens.

Reply:
> Since Non-Null type fields cannot be null, field errors are propagated to be handled by the parent field. If the parent field may be null then it resolves to null, otherwise if it is a Non-Null type, the field error is further propagated to it’s parent field.

So it doesn't say anything about additional field errors being thrown.  It just says that when there's a field error due to non-nullability, it should bubble up as far as necessary.

Reply:
So it seems to me that we list the error as far down the rabbit-hole as possible (whatever first caused the error), then bubble up the null as much as necessary.  Which is how it works now.  If a resolver or serialization error occurs, the error is recorded, and then null is bubbled up without any additional errors listed.  Best I can read the spec, that's what is intended.  Keep in mind the caller knows the nullability graph, and the path of the error, and can deduce from the message why the parent nodes are null.

Reply:
Maybe, maybe...

> That is, only one error should be added to the errors list **per field**.


Per field. This allows you to ambiguously interpret the specification. @benjie Could you help us with understanding spec, please?